test/blackbox/utils/data_generators.cpp

```diff
@@ -81,6 +81,28 @@ std::list<KeyedHelloWorld> default_keyedhelloworld_data_generator(
     return returnedValue;
 }
 
+std::list<KeyedHelloWorld> uniquekey_keyedhelloworld_data_generator(
+        size_t max)
+{
+    uint16_t index = 0;
+    size_t maximum = max ? max : 10;
+    std::list<KeyedHelloWorld> returnedValue(maximum);
+
+    std::generate(returnedValue.begin(), returnedValue.end(), [&index]
+            {
+                KeyedHelloWorld hello;
+                hello.index(index);
+                hello.key(index + 1);
+                std::stringstream ss;
+                ss << "HelloWorld " << index;
+                hello.message(ss.str());
+                ++index;
+                return hello;
+            });
+
+    return returnedValue;
+}
```

Comment:
Maybe, instead of creating the new generator where the only change with respect to the already existing generator is use a different key in each sample, the previous generator can be used giving as parameter flag in case that the key must be unique.

```
std::list<KeyedHelloWorld> default_keyedhelloworld_data_generator(
        size_t max,
        bool unique_key)
```

This is a mere suggestion and I am fine leaving the code as it is.